Neos.Media/Classes/Domain/Repository/ImageVariantRepository.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Media\Domain\Repository;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Media\Domain\Model\ImageVariant;
+use Neos\Media\Domain\ValueObject\Configuration\VariantPreset;
+
+/**
+ * A repository for ImageVariants
+ *
+ * @Flow\Scope("singleton")
+ */
+class ImageVariantRepository extends AssetRepository
+{
+    public function findVariantsByName(?string $identifier, ?string $variantName, int $limit = null): array
+    {
+        $queryBuilder = $this->entityManager->createQueryBuilder()
+            ->select('iv')
+            ->from(ImageVariant::class, 'iv')
+            ->setMaxResults($limit);
+
+        if (is_null($identifier) && is_null($variantName)) {
+            $queryBuilder
+                ->where('iv.presetIdentifier IS NULL')
+                ->andWhere('iv.presetVariantName IS NULL');
+        } else {
+            $queryBuilder
+                ->where('iv.presetIdentifier = (:configuredIdentifiers)')
+                ->setParameter('configuredIdentifiers', $identifier);
+
+            $queryBuilder
+                ->andWhere('iv.presetVariantName = (:configuredVariantName)')
+                ->setParameter('configuredVariantName', $variantName);
+        }
+
+        return $queryBuilder->getQuery()->execute();
+    }
+
+    /**
+     * Returns array of ImageVariants with outdated presets
+     *
+     * @param VariantPreset[] $configuredPresets
+     * @param int|null $limit
+     * @return ImageVariant[]
+     */
+    public function findAllWithOutdatedPresets(array $configuredPresets, int $limit = null): array
```

Comment:
i think $configuredPresets should not be a dependency as parameter but rather 

```
$currentPresets = $this->assetVariantGenerator->getVariantPresets();
```

should be used